addons/account/models/account_account.py

```diff
@@ -858,6 +1089,262 @@ def get_import_templates(self):
     def _merge_method(self, destination, source):
         raise UserError(_("You cannot merge accounts."))
 
+    def action_merge(self):
+        """ Merge the accounts in `self`:
+            - the first one is extended to each company of the accounts in `self`, keeping their codes and names;
+            - the others are deleted; and
+            - journal items and other references are retargeted to the first account.
+
+            If the companies of several accounts overlap, this is an accounting operation, so we check that no impacted entries are in a locked period.
+            If the companies don't overlap, from an accounting perspective, each company still has its own independent view of each account.
+        """
+        # Step 1: Checks.
+        if len(self) < 2:
+            raise UserError(_("You must select at least 2 accounts to merge."))
+
+        for field in ['currency_id', 'deprecated', 'account_type', 'reconcile', 'non_trade']:
+            if len(set(self.mapped(field))) != 1:
+                raise UserError(_(
+                    "You may only merge accounts that have the same account type, currency, deprecated status, "
+                    "reconciliation status, and trade/non-trade receivable status."
+                ))
+
+        # Check that if any companies overlap, no impacted entries are in a locked period.
+        accounts_by_company = defaultdict(lambda: self.env['account.account'])
+        for account in self:
+            for company in account.company_ids:
+                accounts_by_company[company] |= account
+
+        # If exactly one of the accounts contains locked entries, we will merge the other ones into it.
+        account_to_merge_into = self.env['account.account']
+
+        is_irreversible = False
+        merge_issues = defaultdict(dict)
+        for company, accounts in accounts_by_company.items():
+            if len(accounts) > 1:
+                is_irreversible = True
+                user_lock_date = company._get_user_fiscal_lock_date()
+                if user_lock_date and (locked_items := self.env['account.move.line'].sudo().search([
+                    ('account_id', 'in', (accounts - account_to_merge_into).ids),
+                    ('date', '<=', user_lock_date),
```

Comment:
You want the hashed entries, right?
```suggestion
                    ('date', '<=', user_lock_date),
                    ('inalterable_hash', '!=', False),
```

Reply:
yes sir! we do sir!